sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -124,6 +135,17 @@ public void onEnd(ReadableSpan span) {
       return;
     }
     worker.addSpan(span);
+
+    if (worker.spansList.size() >= worker.halfMaxQueueSize) {
```

Comment:
Can this logic live down in the worker itself, rather than spreading it across two places?